### PR TITLE
feat(Offer): added feature flag for ProductSelector

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## What?

- linked `web-onboarding` with the versions of NO/DK onboarding flows that support cross selling (`onboarding-NO-v3`, `onboarding-DK-v2`)
- added a feature flag (`FEATURE_OFFER_PAGE_PRODUCT_SELECTOR`) to conditionally display the new component that enables cross selling.
- updated `QuoteCart` graphql query so it requests `standaloneQuotes`/`additionalQuotes` new fields and `insuranceType` field for possible variations. Graphql types were also proper updated.
- added  quote selectors - `getStandaloneProducts`/`getAdditionalProducts` that parses `Standalone/Additional` quotes into a `Product`. A `Product` describe a content of the cards we'll use to enable cross selling.
- general refactorings under `pages/Offer` folder, like promoting `Product` type and remove unused prop from `PageWrapper`.

## Why?

To support the addition of _Pentagon House Cross Sell_ feature 

**Ticket(s): [GRW-1337](https://hedvig.atlassian.net/browse/GRW-1337)**'

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1121** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1121" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1121?utm_source=stack-comment).

File: src/client/pages/Offer/types.ts
Comment (robinandeer) [inline src/client/pages/Offer/types.ts:16]: Not a big deal, just thinking if it makes to introduce a new "product" entity here - what do you think?

Or is this just another representation of a quote?

**Review by robinandeer — APPROVED** [review]
am I missing something -- I don't find anywhere that you are using the new feature flag?

[](https://app.graphite.dev/macros/HedvigInsurance)

Otherwise it looks good

**Comment by guilhermespopolin** [comment]
> am I missing something -- I don't find anywhere that you are using the new feature flag?
> 
> [](https://app.graphite.dev/macros/HedvigInsurance) [  ](https://app.graphite.dev/macros/HedvigInsurance) [ ](https://app.graphite.dev/macros/HedvigInsurance)
> 
> Otherwise it looks good

Oh, its usage is located down in the stack tree.

File: src/client/pages/Offer/types.ts
Comment (guilhermespopolin) [reply src/client/pages/Offer/types.ts:16]: 🤔 I guess I could update [`OfferQuote`](https://github.com/HedvigInsurance/web-onboarding/blob/master/src/client/pages/Offer/types.ts#L24) type by including a `description` and an `image` and use it in `getStandaloneProducts`/`getAdditionalProducts` selectors.

File: src/client/api/quoteCartQuerySelectors.ts
Comment (alebedev) [inline src/client/api/quoteCartQuerySelectors.ts:108]: Nit: formatting rules may be different between locales.  Should we format on view level with some utils?

**Review by alebedev — APPROVED** [review]

File: src/client/api/quoteCartQuerySelectors.ts
Comment (robinandeer) [reply src/client/api/quoteCartQuerySelectors.ts:108]: Hmm might be a good idea, let's see in test session!

**Comment by robinandeer** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1121?utm_source=stack-comment).


Job Reference: zxLtjZlK3VtfUNAuwuaI

**Pull request merged**
